hibernate-reactive-core/src/test/java/org/hibernate/reactive/schema/SchemaUpdateSchemaNameMySQLTest.java

```diff
@@ -0,0 +1,167 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive.schema;
+
+import static org.hibernate.reactive.containers.DatabaseConfiguration.DBType.MYSQL;
+import static org.hibernate.tool.schema.JdbcMetadaAccessStrategy.GROUPED;
+import static org.hibernate.tool.schema.JdbcMetadaAccessStrategy.INDIVIDUALLY;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
+import javax.persistence.Table;
+
+import org.hibernate.cfg.Configuration;
+import org.hibernate.reactive.BaseReactiveTest;
+import org.hibernate.reactive.provider.Settings;
+import org.hibernate.reactive.testing.DatabaseSelectionRule;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import io.vertx.ext.unit.TestContext;
+
+public abstract class SchemaUpdateSchemaNameMySQLTest extends BaseReactiveTest {
```

Comment:
We have this class already in the project: https://github.com/hibernate/hibernate-reactive/blob/main/hibernate-reactive-core/src/test/java/org/hibernate/reactive/SchemaUpdateMySqlTestBase.java

Do we need a separate class for this additional tests or we can add them to the existing class?

Reply:
It's possible to add the tests to current HR MySQL test but I may need to bring over the the ORM version of the entities which don't have PK/FK defined.  The tests will fails when loading the ASimpleNext metadata during build session factory.

I can do that but I think it muddies the scope of the **SchemaUpdateMySqlTestBase.java**

Reply:
Well OK, it would be nice to have all the common configuration in one place but we can do it later.

Reply:
I think it's worth having this in a separate test because it really is very much a corner case.

Notice that the entities map `schema = "test"`.

For MySQL, `#getNameQualifierSupport` returns `NameQualifierSupport.CATALOG`. 

In other words, the mapping really is incorrect; it should map `catalog="test"`.

Internally, MySQL `DatabaseMetaData` method take the `catalog` value and uses it like it is the schema. That's mentioned [here](https://github.com/hibernate/hibernate-reactive/blob/main/hibernate-reactive-core/src/main/java/org/hibernate/reactive/provider/service/MySqlReactiveInformationExtractorImpl.java#L21-L25)

I don't remember, but I think that this weird behavior was introduced at some point, and old mappings got broken.

This is a test that the legacy mapping will still work.

Reply:
@blafond Can you add some javadoc for this class, please?

From what I understand, we are testing:
1. Combination of catalog + schema names
2. Two different classes with with the same entity name and table name

I also don't see anything specific to MySQL in this test class. I think it will work for MSSQL if we change the name of the catalog to `master` during the build and probably for PostgreSQL without additional changes. @blafond could you also check if it works for the other dbs?